internal/services/license/machineid.go

```
@@ -0,0 +1,78 @@
+	return hex.EncodeToString(sum[:])[:32]
+}
+
+// persistFingerprint best-effort writes the fingerprint. A read-only config dir
+// is not fatal: the fingerprint is deterministic, it just gets recomputed.
+func persistFingerprint(path, fingerprint string) string {
+	if err := os.MkdirAll(filepath.Dir(path), 0755); err != nil {
+		log.Warn().Err(err).Str("path", path).Msg("failed to create fingerprint directory")
+		return fingerprint
+	}
+
+	if err := os.WriteFile(path, []byte(fingerprint), 0644); err != nil {
```

golangci-lint (2.12.2)

[medium] G306: Expect WriteFile permissions to be 0600 or less

(gosec)